File: 1-c/mario-more/mario.c
// Program to print adjacent pyramid of blocks
// Left-aligned and right-aligned
// Height is based on user input (between 1 and 8, inclusive)

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;

    // Keep doing this prompt while user input is < 1 or > 8
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // For each row
    for (int i = 0; i < height; i++)
    {
        // <-Left side
        // On left side for each line, print a space before a brick
        for (int j = 1; j < height - i; j++)
        {
            printf(" ");
        }

        // On left side for each line, print a brick
        for (int k = 0; k < i + 1; k++)
        {
            printf("#");
        }

        // Middle
        // Print two spaces in the middle to separate left and right pyramids
        printf("  ");

        // ->Right side
        // On right side for each line, print a brick
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }

        // Move to next row
        printf("#\n");
    }
}

File: 6-python/sentimental-mario-less/mario.py
# Mario in Python


def main():
    """Keep doing this prompt if user input is < 1 or > 8 or non-numeric."""
    try:
        height = 0
        while height < 1 or height > 8:
            height = int(input("Height: "))
    except ValueError:
        main()

    # Function call to print pyramid
    print_pyramid(height)


def print_pyramid(height):
    """Prints out a left-aligned half-pyramid of a specified height."""

    # For each row, start, stop, step
    # <- Left side
    # On left side for each line, print a space before a brick
    for i in range(1, height + 1, 1):
        print(" " * (height - i), end="")

        # Then print a brick
        print("#" * i)


if __name__ == '__main__':
    main()